build/builder.go

```diff
@@ -998,5 +1024,14 @@ func (e deltaBranchSetError) Error() string {
 	return fmt.Sprintf("repository metadata in shard %q contains a different set of branch names than what was requested, which is unsupported in a delta shard build. old: %+v, new: %+v", e.shardName, e.old, e.new)
 }
 
+type deltaIndexOptionsMismatchError struct {
+	shardName  string
+	newOptions HashOptions
+}
+
+func (e *deltaIndexOptionsMismatchError) Error() string {
+	return fmt.Sprintf("one or more index options for shard %q do not match Builder's index options. These index option updates are incompatible with delta build. New index options: %+v", e.shardName, e.newOptions)
```

Comment:
the error message looks like this:
`one or more index options for shard "exp_indexes_dont_commit/newgit220521_v16.00000.zoekt" do not match Builder's index options. These index option updates are incompatible with delta build. New index options: {sizeMax:1048576 disableCTags:true cTags: cTagsMustSucceed:false largeFiles:[]}`